gcs_to_bq/common/census_to_bq.py

```diff
@@ -0,0 +1,72 @@
+import csv
+import os
+import json
+import time
+from google.cloud import bigquery, storage
+from google.cloud.storage import Blob
+from .json_tools import JsonTable, to_dict
+
+def write_state_names_to_bq(table_name, gcs_bucket, filename):
+  """Writes state names to big query from bucket
+     
+     table_name: The name of the biquery table to write to
+     gcs_bucket: The name of the gcs bucket to read the data from
+     filename: The name of the file in the gcs bucket to read from"""
+  client = bigquery.Client()
+  # TODO replace this with the dataset we'll end up actually using
+  dataset_ref = client.dataset('aaron_census_test')
+  job_config = bigquery.LoadJobConfig()
+  
+  # Always append, so we can keep the history.
+  job_config.write_disposition = 'WRITE_APPEND'
+  
+  job_config.schema = [
+    bigquery.SchemaField('ingestion_time', 'TIMESTAMP'),
+    bigquery.SchemaField('state_code', 'STRING'),
+    bigquery.SchemaField('state_name', 'STRING'),
+  ]
+
+  table = load_data_as_json_table(gcs_bucket, filename)
+  load_job = client.load_table_from_json(
+    table.data,
+    dataset_ref.table(table_name),
+    job_config=job_config)
+
+  load_job.result()  # Wait for table load to complete.
+
+
+def census_json_as_json_table(json):
+  col_names = json[0].copy()
+  data = []
+  for row in json[1:]:
+    new_row = {}
+    for i in range(len(col_names)):
+      new_row[col_names[i]] = row[i]
+    data.append(new_row)
+  return JsonTable(data, col_names)
+
+
+def load_data_as_json_table(gcs_bucket, filename):
+  client = storage.Client()
+  bucket = client.get_bucket(gcs_bucket)
+  blob = bucket.blob(filename)
+  local_path = local_file_path(filename)
+  blob.download_to_filename(local_path)
+
+  with open(local_path) as json_file:
+    data = json.load(json_file)
```

Comment:
nit: this may be redundant given we check for proper json when writing to gcs but if for whatever reason the file is malformatted we could verify that what we load is valid json

Reply:
Good idea, I'll add a check here. Where do we check that when writing to gcs?

Reply:
I added the error handling at the top-level (I think that's generally better practice to avoid poorly defined behavior when a fatal error is swallowed in low-level code)